### ISSUE TITLE
lack of stackable traits

### ISSUE THREAD

**Issue by Flamewh33l** [OpenRA/OpenRA#13594]
i've been doing a bit of modding recently in an effort to find a use for the Forward Command structure in the Red Alert mod that wasn't build radius or vision range; my idea was to make it similar to the Veteran Academy from RA3 where in by you capture the Forward Command and all newly trained units you produce will be granted a rank up from the get-go. this idea can be done with a Prerequisite called 'ArmoryLevelUp' and then change the Prerequisite on the units from the building.upgraded to house that Prerequisite within the trait 'ProducibleWithLevel' but it only does it to rank 1 (although gives you an idea what is after); now it is possible to do it in Lua and work out a way to stack traits from the Yaml so this trick will work... good luck in working that one out and get back to me when progress has been made, i'll attempt to work something out as well so we can keep brainstorming through :)

**Comment by abcdefg30** [comment]
I can think of two (quick) possible solutions for this case:
- Add condition support to `ProducibleWithLevel`
- Create a `GrantPrerequisiteOnCondition` trait

**Comment by reaperrr** [comment]
> Create a GrantPrerequisiteOnCondition trait

Couldn't we just make the existing ProvidesPrerequisite trait conditional? Though it might still be better to rename it if we do.

**Comment by GraionDilach** [comment]
:-1: on renaming ProvPrereq.

**Comment by CombinE88** [comment]
GrantPrerequisiteOnCondition is the easy way to got here

**Comment by pchote** [comment]
I agree with @GraionDilach, there is no need to rename `ProvidesPrerequisite`.

**Comment by forcecore** [comment]
Stackable ProvidesPrequisite would be nice, it would allow natural limiting of # of aircraft units producible depending on the airfields/helipads.

**Comment by pchote** [comment]
We already plan to solve that by reimplementing the prereqs system using the `IntegerExpression` backend.
